runtime/gc_modron_startup/mgcalloc.cpp

```diff
@@ -468,7 +468,11 @@ J9AllocateObject(J9VMThread *vmThread, J9Class *clazz, uintptr_t allocateFlags)
 		env->disableInlineTLHAllocate();
 	}	
 #endif /* J9VM_GC_THREAD_LOCAL_HEAP */	
-
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	if (J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassContainsUnflattenedFlattenables)) {
+		defaultValueWithUnflattenedFlattenables(clazz, objectPtr);
```

Comment:
This must be able to fail, in which case NULL should be returned.

Reply:
Do you mean `defaultValueWithUnflattenedFlattenables(clazz, objectPtr);` has to have a failing mechanism? If yes, I have the following questions: 
1. By what criteria is this method considered as failing? Like the method didn't find a field to assign default value?
2. Currently the method is not returning anything. If returning NULL when failing, what should be returned when not failing?
Thank you very much!

Reply:
@gacholio Or do you mean the function `J9AllocateObject` has to be able to fail?

Reply:
Sorry, I thought the default code was allocating new objects (which could fail) - it appears to be using pre-allocated singletons for the default case, so it can not in fact fail.

Reply:
Thank you very much!